api/proto/attest/attest.proto

```diff
@@ -42,18 +42,44 @@ message ZAttestNonceResp {
   bytes nonce = 1;      //nonce to use in quote generation
 }
 
+//Taken from
+//5.2 Crypto Agile Log Entry Format, from https://trustedcomputinggroup.org/wp-content/uploads/EFI-Protocol-Specification-rev13-160330final.pdf
+message TpmEventLogEntry {
+   uint32 index = 1;           // Sequential index of event in the log
+   uint32 pcr_index = 2;        // PCR index to which this event was measured
+   uint32 event_type = 3;       // The type of this event
+   bytes  digest256 = 4;       // The sha256 digests corresponding to this event
+   bytes  event_data_binary = 5; // The raw event data bytes
+   string event_data_string = 6; // Textual representation of the event data
+}
+
+message GPSCoordinates {
+   double latitude = 1;  // Latitude of the device geo-location
+   double longitude = 2; // Longitude of the device geo-location
+}
+
 message ZAttestQuote {
   bytes attestData = 1;  //TPMS_ATTEST (Table 2:123) in https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf
-                          //nonce is included in attestData
+                         //nonce is included in attestData
   bytes signature = 2;   //signature to verify attestData
+  repeated bytes pcr_sha256_values = 3;     // Current values of TPM PCRs 0-15, for sha256 bank
+  repeated TpmEventLogEntry event_log = 4;  // TPM Event log
+  string eve_version = 5;                   // Current EVE version
+  string fw_version = 6;                    // Current Firmware version
```

Comment:
Wonder whether we should generalize this to a repeated message, where message has a versionType and a version string, and define EVE and BIOS as the initial versionTypes. That way we can easily later extend to include other version strings which might matter.

Reply:
First of all, I like this idea @eriknordmark -- but I'm also wondering what exact purpose do these version string serve? The way it appears to me by reading API docs this is simply to map hashes to a human readable string, correct? I'd like to understand semantics of this better before we add it to the API @cshari-zededa

Reply:
> First of all, I like this idea @eriknordmark -- but I'm also wondering what exact purpose do these version string serve? The way it appears to me by reading API docs this is simply to map hashes to a human readable string, correct? I'd like to understand semantics of this better before we add it to the API @cshari-zededa

Hi @rvs, I am adding these versions in the attestation request, so that,  Controller can look up the expected measurements for these reported versions, and match the measurements with measurements in the attestation request(i.e. the PCR values). To start with, we will have UEFI and EVE versions reported, but as @eriknordmark suggested, I am modifying it to be a repeated list, so that we can append this list with any future software layer as well (like bootloader version etc).